tensorflow/core/kernels/segment_reduction_ops_gpu.cu.h

```diff
@@ -314,14 +322,15 @@ __device__ T ReduceBlockAlongCols(ReduceOp reduce_op, const T& value,
 // size (as long as it spans at least the maximum value in indices). This also
 // applies to the weights vector.
 template <typename Treducevec, typename Tvec, typename Tindex,
-          typename Tsegmentids, typename ReduceOp, typename Tinit>
+          typename Tindices, typename Tsegmentids, typename ReduceOp,
+          typename Tinit, typename Tweight>
 __global__ void SegmentReduceVectorKernel(
     Tindex nouter, Tindex ninner_vec, Tsegmentids nsegments, ReduceOp reduce_op,
     Tinit initial_value, Tinit empty_segment_value, bool is_mean, bool is_sqrtn,
     const Tvec* __restrict__ input_vec,          // [nouter or any, ninner_vec]
     const Tindex* __restrict__ segment_offsets,  // [nsegments + 1]
-    const Tindex* __restrict__ indices,          // [nouter] (optional)
-    const Tinit* __restrict__ weights,           // [nouter or any] (optional)
+    const Tindices* __restrict__ indices,        // [nouter] (optional)
+    const Tweight* __restrict__ weights,         // [nouter or any] (optional)
```

Comment:
nit: we seem to be mixing singular/plural here.  Can we add some consistency?

Reply:
Done, renamed `Tweight` to `Tweights`.